tests/integration/integration_go_test.go

```diff
@@ -1221,13 +1221,41 @@ func TestParameterizedGo(t *testing.T) {
 		PrePrepareProject: func(info *engine.Projinfo) error {
 			e := ptesting.NewEnvironment(t)
 			e.CWD = info.Root
+
+			// Overwrite the main program with the actual test program that uses the generated SDK.
+			actualProgram, err := os.ReadFile(filepath.Join(e.CWD, "actual_program.txt"))
+			require.NoError(t, err)
+			e.WriteTestFile("main.go", string(actualProgram))
+
+			// Generate the SDK for the provider.
 			path := info.Proj.Plugins.Providers[0].Path
 			_, _ = e.RunCommand("pulumi", "package", "gen-sdk", path, "pkg", "--language", "go")
-			return nil
+
+			// Add a reference to the generated SDK in go.mod.
+			return appendLines(filepath.Join(e.CWD, "go.mod"), []string{
+				"require example.com/pulumi-pkg/sdk/go v1.0.0",
+				"replace example.com/pulumi-pkg/sdk/go => ./sdk/go",
```

Comment:
We could also use [AddReplace/AddRequire](https://pkg.go.dev/golang.org/x/mod@v0.24.0/modfile#File.AddReplace) here, although I think this is fine.

Reply:
Good suggestion! I think I'll keep it as-is just because it's slightly simpler this way.